Competition2017/src/competition/subsystems/drive/commands/DriveForDistanceCommand.java

```diff
@@ -26,14 +27,20 @@
 
     private double previousPositionInches;
     
+    private final PIDManager headingDrivePid;
+    private ContiguousHeading targetHeading;
+    public final double defaultPValue = 1/80d;
+
     private int onTargetCount = 0;
     
     @Inject
     public DriveForDistanceCommand(DriveSubsystem driveSubsystem, XPropertyManager propManager) {
         this.driveSubsystem = driveSubsystem;
         this.requires(driveSubsystem);
         this.travelManager = new PIDManager("Drive to position", propManager, 0.1, 0, 0, 0.5, -0.5);
-        
+
+        headingDrivePid = new PIDManager("HeadingModule", propManager, defaultPValue, 0, 0);
+        targetHeading = new ContiguousHeading();
         onTargetCountThresholdProp = propManager.createPersistentProperty("DrvToPos min stabilization time", 3);
```

Comment:
I'm not sure it's the **most** important thing to make it time, though I agree it would be more robust in theory. The loop counts in practice have worked very well and is a lot simpler to initially write and validate the logic thereof (came about while testing the command and having very poor accuracy when the command completed). Case in point, our current PID logic has no concept of time even though in theory it should to accurately compute the derivative and accumulated errors for instance.

There's a task to fold the concept of 'on target for time' into the PIDManager and I agree it probably makes sense to have it to be `time` based there.